creator-node/src/snapbackSM/snapbackSM.js

```diff
@@ -1208,8 +1228,15 @@ class SnapbackSM {
       // Log decision tree
       this._printStateMachineQueueDecisionTree(decisionTree, jobId)
 
-      // Increment and adjust current slice by this.moduloBase
-      this.currentModuloSlice = (this.currentModuloSlice + 1) % this.moduloBase
+      if (useModulo) {
+        // Increment and adjust current slice by this.moduloBase
+        this.currentModuloSlice =
+          (this.currentModuloSlice + 1) % this.moduloBase
+      } else {
+        // The next job should start processing where this one ended or loop back around to the first user
+        const lastProcessedUser = nodeUsers.pop() || { user_id: 0 }
```

Comment:
sweet - even tho this is end of function, would prefer not to mutate `nodeUsers` here - instead just fetch last value